xbmc/pvr/PVRThumbLoader.cpp

```diff
@@ -102,35 +99,9 @@ bool CPVRThumbLoader::FillThumb(CFileItem& item)
 
 std::string CPVRThumbLoader::CreateChannelGroupThumb(const CFileItem& channelGroupItem)
 {
-  const CPVRGUIDirectory channelGroupDir(channelGroupItem.GetPath());
-  CFileItemList channels;
-  if (channelGroupDir.GetChannelsDirectory(channels))
-  {
-    std::vector<std::string> channelIcons;
-    for (const auto& channel : channels)
-    {
-      const std::string& icon = channel->GetArt("icon");
-      if (!icon.empty())
-        channelIcons.emplace_back(CPVRCachedImages::UnwrapImageURL(icon));
-
-      if (channelIcons.size() == 9) // limit number of tiles
-        break;
-    }
-
-    std::string thumb = StringUtils::Format(
-        "{}?ts={}", // append timestamp to Thumb URL to enforce texture refresh
-        CTextureUtils::GetWrappedImageURL(channelGroupItem.GetPath(), "pvr"), std::time(nullptr));
-    const std::string relativeCacheFile = CTextureCache::GetCacheFile(thumb) + ".png";
-    if (CPicture::CreateTiledThumb(channelIcons, CTextureCache::GetCachedPath(relativeCacheFile)))
-    {
-      CTextureDetails details;
-      details.file = relativeCacheFile;
-      details.width = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_imageRes;
-      details.height = details.width;
-      CServiceBroker::GetTextureCache()->AddCachedTexture(thumb, details);
-      return thumb;
-    }
-  }
+  std::string thumb = StringUtils::Format(
+      "{}?ts={}", // append timestamp to Thumb URL to enforce texture refresh
+      CTextureUtils::GetWrappedImageURL(channelGroupItem.GetPath(), "pvr"), std::time(nullptr));
 
-  return {};
+  return thumb;
```

Comment:
Maybe we can get rid off the variable here and simply return the stringutils::format result.

Reply:
updated